cvetools/cvesearch.go

```diff
@@ -204,16 +204,30 @@ func (cv *CveTools) ScanAppPackage(req *share.ScanAppRequest, namespace string)
 
 // ScanImage helps the Image scanning
 func (cv *CveTools) ScanImage(ctx context.Context, req *share.ScanImageRequest, imgPath string) (*share.ScanResult, error) {
-	var err error
 	result := &share.ScanResult{
-		Provider:        share.ScanProvider_Neuvector,
-		Version:         cv.CveDBVersion,
-		CVEDBCreateTime: cv.CveDBCreateTime,
-		Error:           share.ScanErrorCode_ScanErrNone,
-		Registry:        req.Registry,
-		Repository:      req.Repository,
-		Tag:             req.Tag,
-		Layers:          make([]*share.ScanLayerResult, 0),
+		Provider:           share.ScanProvider_Neuvector,
+		Version:            cv.CveDBVersion,
+		CVEDBCreateTime:    cv.CveDBCreateTime,
+		Error:              share.ScanErrorCode_ScanErrNone,
+		Registry:           req.Registry,
+		Repository:         req.Repository,
+		Tag:                req.Tag,
+		Layers:             make([]*share.ScanLayerResult, 0),
+		ScanTypesRequested: req.ScanTypesRequested,
+	}
+
+	log.WithFields(log.Fields{"scanTypesRequested": req.ScanTypesRequested}).Info("Scan types requested")
```

Comment:
req.ScanTypesRequested is a pointer. Are you sure this line can print out the scan type value, not the value's pointer?

Reply:
I was using this line to check whether the field was nil when debugging the controller request. I will move and modify this to be more generally helpful/correct.